auth0/resource_auth0_tenant.go

```diff
@@ -276,9 +277,13 @@ func readTenant(d *schema.ResourceData, m interface{}) error {
 	d.Set("support_email", t.SupportEmail)
 	d.Set("support_url", t.SupportURL)
 	d.Set("allowed_logout_urls", t.AllowedLogoutURLs)
-	d.Set("session_lifetime", t.SessionLifetime)
+	if t.SessionLifetime != nil {
+		d.Set("session_lifetime", t.SessionLifetime)
+	}
```

Comment:
Oh, you merged before I was able to respond. If I remember correctly, the issue is that a `schema.TypeInt` is cast to zero when not provided, rather than simply not being added to the request. That's not caught in [this test](https://github.com/alexkappa/terraform-provider-auth0/blob/master/auth0/resource_auth0_tenant_test.go#L63) because the `session_lifetime` parameter is set. If that is set to `nil` then the tests _should_ fail without this change.

Reply:
Yes, sorry. I did that in order to merge/release before the weekend was over cause its the only time I have for this project these days.

So if I understand correctly, it's caused in the case `session_lifetime` wasn't in the tenant settings when creating the tenant right? Since the tenant is a special thing -we can't create one- we might need to do include your changes after all, but since its not common within the provider to do these nil checks I preferred to leave it out. Instead, the mechanism used is during expansion of terraform config to auth0 structs we check for `HasChage`, `IsNewResource` etc.

Reply:
> its the only time I have for this project these days.

Totally understandable. Thank you for volunteer work, I'm sure the community appreciates it (I certainly do).

> it's caused in the case `session_lifetime` wasn't in the tenant settings when creating the tenant right?

Specifically, that it's not included in the `auth0_tenant`'s arguments when creating the resource.

> the mechanism used is during expansion of terraform config to auth0 structs we check for HasChage, IsNewResource etc.

Oh, I see. Taking a look at https://github.com/alexkappa/terraform-provider-auth0/blob/15de64a52bc1b430dfc80b525f369959e68a8884/auth0/resource_auth0_tenant.go#L316-L318

perhaps the only change necessary is to amend the `SessionLifetime` line to:

```go
SessionLifetime:     Int(d, "session_lifetime", IsNewResource(), HasChange()),
```

I believe I added the `nil` check to `idle_session_lifetime` simply for "safety" (I don't really know what I'm doing here, as you can tell).

If I have some time today, I'll try that out.